plugins/modules/s3key_info.py

```diff
@@ -0,0 +1,211 @@
+DOC_DIRECTORY = 'user-management'
+STATES = ['info']
+OBJECT_NAME = 'S3 Key'
+
+OPTIONS = {
+    'user_id': {
+        'description': ['The ID of the user'],
+        'available': STATES,
+        'required': STATES,
+        'type': 'str',
+    },
+    'api_url': {
+        'description': ['The Ionos API base URL.'],
+        'version_added': '2.4',
+        'env_fallback': 'IONOS_API_URL',
+        'available': STATES,
+        'type': 'str',
+    },
+    'username': {
+        # Required if no token, checked manually
+        'description': ['The Ionos username. Overrides the IONOS_USERNAME environment variable.'],
+        'aliases': ['subscription_user'],
+        'required': STATES,
+        'env_fallback': 'IONOS_USERNAME',
+        'available': STATES,
+        'type': 'str',
+    },
+    'password': {
+        # Required if no token, checked manually
+        'description': ['The Ionos password. Overrides the IONOS_PASSWORD environment variable.'],
+        'aliases': ['subscription_password'],
+        'required': STATES,
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_PASSWORD',
+        'type': 'str',
+    },
+    'token': {
+        # If provided, then username and password no longer required
+        'description': ['The Ionos token. Overrides the IONOS_TOKEN environment variable.'],
+        'available': STATES,
+        'no_log': True,
+        'env_fallback': 'IONOS_TOKEN',
+        'type': 'str',
+    },
+}
+
+def transform_for_documentation(val):
+    val['required'] = len(val.get('required', [])) == len(STATES) 
+    del val['available']
+    del val['type']
+    return val
+
+DOCUMENTATION = '''
+---
+module: s3key_info
+short_description: List Ionos Cloud S3Keys of a given user.
+description:
+     - This is a simple module that supports listing S3Keys.
+version_added: "2.0"
+options:
+''' + '  ' + yaml.dump(yaml.safe_load(str({k: transform_for_documentation(v) for k, v in copy.deepcopy(OPTIONS).items()})), default_flow_style=False).replace('\n', '\n  ') + '''
+requirements:
+    - "python >= 2.6"
+    - "ionoscloud >= 6.0.2"
+author:
+    - "IONOS Cloud SDK Team <sdk-tooling@ionos.com>"
+'''
+
+EXAMPLES = '''
+    - name: List S3Keys for user
+      s3key_info:
+        user_id: "{{ user_id }}"
+        register: s3key_info_response
+
+    - name: Show S3Keys
+      debug:
+        var: s3key_info_response.result
+'''
+
+
+def get_s3keys(module, client):
+    user_id = module.params.get('user_id')
+    user_s3keys_server = ionoscloud.UserS3KeysApi(client)
+
+    try:
+        results = []
+        for s3key in user_s3keys_server.um_users_s3keys_get(user_id).items:
+            results.append(s3key.to_dict())
+        return {
+            'action': 'info',
+            'changed': False,
+            's3keys': results
+        }
+
+    except Exception as e:
+        module.fail_json(msg="failed to list the s3keys: %s" % to_native(e))
+        return {
```

Comment:
The **return** can be ommited. The program will stop the execution at **module.fail_json**

Reply:
thanks for noticing!